specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json

```diff
@@ -1737,1525 +1737,2518 @@
+          },
+          {
+            "name": "galleryInVMAccessControlProfile",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GalleryInVMAccessControlProfile"
+            },
+            "description": "Parameters supplied to the create or update gallery inVMAccessControlProfile operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GalleryInVMAccessControlProfile"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/GalleryInVMAccessControlProfile"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common-types/v1/common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create or update a gallery inVMAccessControlProfile.": {
+            "$ref": "./examples/galleryResourceProfileExamples/GalleryInVMAccessControlProfile_Create.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "GalleryInVMAccessControlProfiles"
+        ],
+        "operationId": "GalleryInVMAccessControlProfiles_Update",
+        "description": "Update a gallery inVMAccessControlProfile.",
+        "parameters": [
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "galleryName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Shared Image Gallery in which the InVMAccessControlProfile is to be updated."
+          },
+          {
+            "name": "inVMAccessControlProfileName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the gallery inVMAccessControlProfile to be updated. The allowed characters are alphabets and numbers with dots, dashes, and periods allowed in the middle. The maximum length is 80 characters."
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "galleryInVMAccessControlProfile",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GalleryInVMAccessControlProfileUpdate"
+            },
+            "description": "Parameters supplied to the update gallery inVMAccessControlProfile operation."
+          }
+        ],
+        "responses": {
```

Comment:
Pl refer to: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-1    for guidelines on sync/async completion of a PATCH call...

Reply:
200 for sync, 202 for async... I get that.

but considering we have always returned 200 [(e.g)](https://github.com/Azure/azure-rest-api-specs/blob/462bbcd69b3e8c9f211df7523ccab9665f08153f/specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2024-03-03/gallery.json#L683C10-L683C19) for our LRO patch APIs, it would not be possible to change this behaviour now. hence, we are requesting for the suppression

Reply:
in fact, we had a swagger review for this new API ~2 months back in #29852 . neither the CI nor the reviewer raised this concern, so this behaviour in our service code is already in production

Reply:
1. Do you know if your SDK tracks an async PATCH properly ? It may be the case that PATCH is completing Sync most of the time so terminating with a 200 and No need to use the LRO tracking...

2. Even if the SDK tracks properly for 200+async behaviour - how will consumers of direct REST API work ? Normally there is no LRO with 200 - so this is confusing

@pratikjoshi14 @sorabhtandon   -- what do you folks reckon ?

Reply:
I can check on 1.

For 2. if direct REST API is used, users will need to check headers for Azure-AsyncOperation, regardless of if it is 200 or 202 right?

Reply:
Semantically, since 200 is terminal, I dont think it makes sense for the client to check the headers to try and poll...